pkg/infrastructure/gcp/clusterapi/firewallrules.go

```diff
@@ -198,6 +225,27 @@ func createFirewallRules(ctx context.Context, in clusterapi.InfraReadyInput, net
 		return err
 	}
 
+	// etcd are needed for master communication for etcd nodes
+	firewallName = fmt.Sprintf("%s-etcd", in.InfraID)
+	srcTags = []string{masterTag}
+	targetTags = []string{masterTag}
+	srcRanges = []string{}
+	if err := addFirewallRule(ctx, firewallName, network, projectID, getEtcdPorts(), srcTags, targetTags, srcRanges); err != nil {
+		return err
+	}
+
+	// health checks are needed for masters
+	firewallName = fmt.Sprintf("%s-health-checks", in.InfraID)
+	srcTags = []string{}
+	targetTags = []string{masterTag}
+	srcRanges = []string{"35.191.0.0/16", "130.211.0.0/22"}
```

Comment:
Maybe it's obvious but just in case: let's add a comment here explaining what these IPs represent and/or where they are coming from.